src/main/java/com/adobe/aem/compgenerator/utils/DialogUtils.java

```diff
@@ -69,11 +72,56 @@ public static void createDialogXml(final GenerationConfig generationConfig, fina
         }
     }
 
+    /**
+     * Creates the tab dialog properties.
+     *
+     * @param doc The {@link Document} object
+     * @param rootElement the root element
+     * @param tabs The {@link Tab} object
+     */
+    private static void createDialogProperties(Document doc, Element rootElement, List<Tab> tabs,
+            List<Property> properties) {
+
+        if (null != properties && !properties.isEmpty()) {
+            if (null != tabs && !tabs.isEmpty()) {
+                Node currentNode = createDefaultTabNodeStructure(doc, rootElement);
+                for (Tab tab : tabs) {
+                    Node tabNode = createTabStructure(doc, tab, currentNode);
+                    List<Property> sortedProperties = new ArrayList<>();
+                    
+                    tab.getFields().stream().filter(field -> properties.stream().anyMatch(
+                            property -> property.getField().equals(field) ? sortedProperties.add(property) : false))
+                            .collect(Collectors.toList());
```

Comment:
Is this the correct usage of `.collect()`?  I would expect this to be used to return a list into a variable - i.e. `tabProperties = tab.getFields().stream().filter...().collect(Collectors.toList())`

Reply:
To get in the sorted order, i have used a different list, I have updated the logic to use a different sorting technique, please review.